File: Java/civitas/src/civitas/SorpresaIraCarcel.java
package civitas;

import java.util.ArrayList;

public class SorpresaIraCarcel extends Sorpresa{
    private Tablero tablero;
    SorpresaIraCarcel(Tablero _tablero){
        super("Ir a Carcel");
        tablero = _tablero;
    }
    
    @Override
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if(this.jugadorCorrecto(actual, todos)){
            this.informe(actual, todos);
            todos.get(actual).encarcelar(tablero.getCarcel());
        }
    }
    
    @Override
    public String toString(){
        return "Sorpresa{ " + super.getTexto() + " }";            
    }
}

File: Ruby/civitas/pkg/civitas-0.0.1/lib/sorpresa_ir_a_carcel.rb
#encoding: utf­8
require_relative "sorpresa.rb"
module Civitas
class SorpresaIrACarcel < Sorpresa
  def initialize(_tablero)
    super("Ir a carcel")
    @tablero = _tablero
  end
  
  def aplicar_a_jugador(_actual, _todos)
    if(jugador_correcto(_actual, _todos))
      informe(_actual, _todos)
      _todos.at(_actual).encarcelar(@tablero.numCasillaCarcel)
    end
  end
  
  
  def to_ss()
    return @texto
  end
end
end